x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/rest/action/profile/RestSuggestProfilesAction.java

```diff
@@ -94,6 +96,19 @@ protected RestChannelConsumer innerPrepareRequest(RestRequest request, NodeClien
         );
     }
 
+    @Override
+    protected Exception checkFeatureAvailable(RestRequest request) {
+        final Exception failedFeature = super.checkFeatureAvailable(request);
+        if (failedFeature != null) {
+            return failedFeature;
+        }
+        if (Security.USER_PROFILE_COLLABORATION_FEATURE.check(licenseState)) {
+            return null;
+        } else {
+            return LicenseUtils.newComplianceException(Security.USER_PROFILE_COLLABORATION_FEATURE.getName());
+        }
+    }
```

Comment:
I decided to add the license enforcement directly in the REST handlers instead of creating a base class for it. Because most user profile APIs are not licensed, a base class can possibly get mis-used. I am also applying rule-of-3 here, code repitition under 3 is acceptable and we can hold on extracting common code.